framework/src/utils/MooseUtils.C

```diff
@@ -575,6 +575,34 @@ wildCardMatch(std::string name, std::string search_string)
     return false;
 }
 
+template <>
+int
+convert<int>(const std::string & str)
+{
+  std::stringstream ss(str);
+  int val;
+  if ((ss >> val).fail() || !ss.eof())
```

Comment:
I prefer returning early for simple, short cases because it helps reduce nested scope/indentation levels and make it easier (for me at least) to follow the code logic - so I don't have to keep a stack of conditions that have fired in my working memory.  But this is just a general thought - you are welcome to leave the code as is.

Reply:
These are in this order on purpose.  `ss >> val` reads in... `!ss.eof()` makes sure that we completely read what was in `ss` (i.e. that there is nothing left so that `10dog`) is failure)

But maybe I'm misunderstanding what you're saying.

Reply:
He's saying you should switch the if/else clauses around and have two returns. I can see his point, but I also think having several exits from a routine has other drawbacks.